### PR TITLE
Support for interstitial mediation in iOS

### PR DESCRIPTION
**Pull request by rampr**
Add support for
- Facebook
- Chartboost
- Admob

TODO
- Applovin

### CODE REVIEW

File: ios/config.json [hunk 1]
```diff
@@ -30,14 +30,22 @@
...
+    "Chartboost.framework",
+    "SUFacebookAdapter.framework",
+    "FBAudienceNetwork.framework",
+    "SUAdMobAdapter.framework",
+    "GoogleMobileAds.framework"
   ],
   "additionalLinkerFlags": [
-    "-ObjC"
+    "-ObjC",
+    "-fobjc-arc"
```
Comment (rampr) [inline ios/config.json:48]: needed for admob support

File: ios/SupersonicPlugin.mm [hunk 2]
```diff
@@ -117,7 +134,7 @@ - (void)supersonicRVInitSuccess {
 // This method is invoked if initialization failed, in which
 // case check out 'error'.
 - (void)supersonicRVInitFailedWithError:(NSError *)error {
-    NSLog(@"%s", __PRETTY_FUNCTION__);
+    NSLog(@"%s%@", __PRETTY_FUNCTION__, error);
```
Comment (rampr) [inline ios/SupersonicPlugin.mm:137]: print error message too

**Comment by rohanhc** [comment]
LGTM

**Comment by abhishekit29** [comment]
LGTM

**Comment by sksethi25** [comment]
Looks good to me

File: ios/SupersonicPlugin.mm [hunk 3]
```diff
@@ -184,4 +201,65 @@ - (void)supersonicRVAdStarted {
...
+* Called each time the Interstitial window has opened successfully.
+**/
+-(void)supersonicISShowSuccess{
+    NSLog(@"{supersonic} onInterstitialShown");
+}
+/**
+* Called if showing the Interstitial for the user has failed.
+* You can learn about the reason by examining the ‘error’ value
+**/
+-(void)supersonicISShowFailWithError:(NSError *)error{ }
```
Comment (bijoshtj) [inline ios/SupersonicPlugin.mm:238]: better to add NSLog here as well.

**Pull request merged**